src/QmlControls/ParameterEditorController.cc

```diff
@@ -374,61 +371,88 @@ bool ParameterEditorController::_shouldShow(Fact* fact) const
-                    if (re.isValid()) {
-                        if (!fact->name().contains(re) &&
-                                !fact->shortDescription().contains(re) &&
-                                !fact->longDescription().contains(re)) {
-                            matched = false;
-                        }
-                    } else {
-                        if (!fact->name().contains(searchItem, Qt::CaseInsensitive) &&
-                                !fact->shortDescription().contains(searchItem, Qt::CaseInsensitive) &&
-                                !fact->longDescription().contains(searchItem, Qt::CaseInsensitive)) {
-                            matched = false;
-                        }
+void ParameterEditorController::_performThreadedSearch(void)
+{
+    QStringList rgSearchStrings = searchText().split(' ', Qt::SkipEmptyParts);
+    _searchResults = new QmlObjectListModel;
+
+    for (const QString &paramName: _parameterMgr->parameterNames(_vehicle->defaultComponentId())) {
+        Fact* fact = _parameterMgr->getParameter(_vehicle->defaultComponentId(), paramName);
+        bool matched = _shouldShow(fact);
+        // All of the search items must match in order for the parameter to be added to the list
+        if (matched) {
+            for (const auto& searchItem : rgSearchStrings) {
+                if (_cancelSearchThread) {
+                    delete _searchResults;
+                    return;
+                }
+                QRegularExpression re = QRegularExpression(searchItem, QRegularExpression::CaseInsensitiveOption);
+                if (re.isValid()) {
+                    if (!fact->name().contains(re) &&
+                            !fact->shortDescription().contains(re) &&
+                            !fact->longDescription().contains(re)) {
+                        matched = false;
+                    }
+                } else {
+                    if (!fact->name().contains(searchItem, Qt::CaseInsensitive) &&
+                            !fact->shortDescription().contains(searchItem, Qt::CaseInsensitive) &&
+                            !fact->longDescription().contains(searchItem, Qt::CaseInsensitive)) {
+                        matched = false;
                     }
                 }
             }
-            if (matched) {
-                _searchParameters.append(fact);
-            }
         }
+        if (matched) {
+            _searchResults->append(fact);
+        }
+    }
 
-        _searchParameters.endReset();
-
-        if (_parameters != &_searchParameters) {
-            _parameters = &_searchParameters;
-            emit parametersChanged();
+    _searchResults->moveToThread(this->thread());
+    emit searchResultsReady();
+}
 
-            _currentCategory    = nullptr;
-            _currentGroup       = nullptr;
-        }
+void ParameterEditorController::_swapSearchResults()
+{
+    if (_searchResults) {
+        //delete _parameters;
```

Comment:
@HTRamsey If I comment this out it all works, but leaks. If I put it in it will crash with memory corruption.

Reply:
This might be one of those tasks that's easier to do through QtConcurrent, but yeah I'll take a look at it later.